styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesPodEventTranslator.java

```diff
@@ -131,101 +131,118 @@ public TerminationLogMessage(
     }
   }
 
-  // TODO: fix NPath complexity
   static List<Event> translate(
       WorkflowInstance workflowInstance,
       RunState state,
       Pod pod,
       Stats stats) {
 
     final List<Event> generatedEvents = Lists.newArrayList();
```

Comment:
instead of mutating this shared list seems we could just have `handleError`, `handleStarted` and `handleExited` return `List<Event>` directly.

Then this code could be:

```java
final Optional<Event> hasError = isInErrorState(workflowInstance, pod, mainContainerStatusOpt);
if (hasError.isPresent()) {
  return handleError(state, generatedEvents, hasError.get());
}

if (isStarted(pod, mainContainerStatusOpt)) {
  return handleStarted(workflowInstance, state, generatedEvents);
}

if (isExited(pod, mainContainerStatusOpt)) {
  return handleExited(workflowInstance, state, pod, mainContainerStatusOpt, stats, generatedEvents);
}
```

Reply:
sounds good